Signal/src/ViewControllers/UpdateGroupViewController.m

```diff
@@ -0,0 +1,504 @@
+#import "AddToGroupViewController.h"
+#import "BlockListUIUtils.h"
+#import "ContactTableViewCell.h"
+#import "ContactsViewHelper.h"
+#import "Environment.h"
+#import "GroupViewHelper.h"
+#import "OWSAnyTouchGestureRecognizer.h"
+#import "OWSContactsManager.h"
+#import "OWSTableViewController.h"
+#import "SecurityUtils.h"
+#import "SignalAccount.h"
+#import "SignalKeyingStorage.h"
+#import "TSOutgoingMessage.h"
+#import "UIUtil.h"
+#import "UIView+OWS.h"
+#import "UIViewController+OWS.h"
+#import <SignalServiceKit/NSDate+millisecondTimeStamp.h>
+#import <SignalServiceKit/OWSMessageSender.h>
+#import <SignalServiceKit/TSGroupModel.h>
+#import <SignalServiceKit/TSGroupThread.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface UpdateGroupViewController () <UIImagePickerControllerDelegate,
+    UITextFieldDelegate,
+    ContactsViewHelperDelegate,
+    GroupViewHelperDelegate,
+    AddToGroupViewControllerDelegate,
+    OWSTableViewControllerDelegate,
+    UINavigationControllerDelegate>
+
+@property (nonatomic, readonly) OWSMessageSender *messageSender;
+@property (nonatomic, readonly) ContactsViewHelper *contactsViewHelper;
+@property (nonatomic, readonly) GroupViewHelper *groupViewHelper;
+
+@property (nonatomic, readonly) OWSTableViewController *tableViewController;
+@property (nonatomic, readonly) UIImageView *avatarView;
+@property (nonatomic, readonly) UITextField *groupNameTextField;
+
+@property (nonatomic, nullable) UIImage *groupAvatar;
+@property (nonatomic, nullable) NSSet<NSString *> *previousMemberRecipientIds;
+@property (nonatomic, nullable) NSMutableSet<NSString *> *memberRecipientIds;
+
+@property (nonatomic) BOOL hasUnsavedChanges;
+
+@end
+
+#pragma mark -
+
+@implementation UpdateGroupViewController
+
+- (instancetype)init
+{
+    self = [super init];
+    if (!self) {
+        return self;
+    }
+
+    [self commonInit];
+
+    return self;
+}
+
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder
+{
+    self = [super initWithCoder:aDecoder];
+    if (!self) {
+        return self;
+    }
+
+    [self commonInit];
+
+    return self;
+}
+
+- (void)commonInit
+{
+    _messageSender = [Environment getCurrent].messageSender;
+    _contactsViewHelper = [ContactsViewHelper new];
+    _contactsViewHelper.delegate = self;
+    _groupViewHelper = [GroupViewHelper new];
+    _groupViewHelper.delegate = self;
+
+    self.memberRecipientIds = [NSMutableSet new];
+}
+
+#pragma mark - View Lifecycle
+
+- (void)loadView
+{
+    [super loadView];
+
+    OWSAssert(self.thread);
+    OWSAssert(self.thread.groupModel);
+    OWSAssert(self.thread.groupModel.groupMemberIds);
+
+    [self.memberRecipientIds addObjectsFromArray:self.thread.groupModel.groupMemberIds];
+    self.previousMemberRecipientIds = [NSSet setWithArray:self.thread.groupModel.groupMemberIds];
+
+    self.title = NSLocalizedString(@"EDIT_GROUP_DEFAULT_TITLE", @"The default title for the 'update group' view.");
+    self.navigationItem.leftBarButtonItem =
```

Comment:
I *do* like the "confirm before leaving" 👍 

But this breaks the interactive pop gesture (e.g. swipe to go back).  

Maybe "editing a group" should be a modal activity anyway, rather than pushed, since we're not really "drilling down".

That would solve the problem of people thinking they can "swipe back".